crates/wasmtime/src/module/registry.rs

```diff
@@ -274,17 +340,15 @@ impl GlobalFrameInfo {
 
     /// Registers a new region of code, described by `(start, end)` and with
     /// the given function information, with the global information.
-    fn register(&mut self, start: usize, end: usize, module: &Arc<CompiledModule>) {
-        let info = self
-            .ranges
-            .entry(end)
-            .or_insert_with(|| GlobalModuleFrameInfo {
-                module: ModuleFrameInfo {
-                    start,
-                    module: module.clone(),
-                },
-                references: 0,
-            });
+    fn register(&mut self, start: usize, end: usize, module: &Module) {
+        let info = self.0.entry(end).or_insert_with(|| GlobalRegisteredModule {
+            module: RegisteredModule {
+                start,
+                module: module.compiled_module().clone(),
+                signatures: module.shared_signatures().clone(),
```

Comment:
I'm a little wary of this too, there's no need for the global registration to hang on to signatures, right?